mountpoint-s3-fs/src/upload/incremental.rs

```diff
@@ -133,6 +132,19 @@ where
     }
 }
 
+/// Events emitted by the background queue.
+#[derive(Debug)]
+enum AppendUploadEvent<E> {
+    /// The algorithm used to compute checksums.
+    /// Only emitted once, before any [AppendUploadEvent::PutResponse] event.
+    ChecksumAlgorithm(Option<ChecksumAlgorithm>),
```

Comment:
Couldn't we receive checksum "synchronously" in `AppendUploadQueue::new`? Probably that would mean a delay on `open`, is that the reason for making this request in `run_append_loop`?

Reply:
Yes, we do not want to add latency to `open` (not changed in this PR).